### ISSUE TITLE
set "-P:scalajs:mapSourceURI" option in scala-js build

### ISSUE THREAD

**Issue by xuwei-k** [etorreborre/specs2#590]
Scala-js include your local file path without this options! (e.g. `file:/Users/etorreborre/projects/specs2/specs2-fp-lib-on-js`)

(stacktrace from https://github.com/etorreborre/specs2/issues/589 )

```
[error]   org.specs2.fp.Need$$anon$2.value0$lzycompute(/Users/kenji/specs2-4/target/scala-2.12/file:/Users/etorreborre/projects/specs2/specs2-fp-lib-on-js/fp/shared/src/main/scala/org/specs2/fp/Name.scala:30:42)
[error]   org.specs2.fp.Need$$anon$2.value0(/Users/kenji/specs2-4/target/scala-2.12/file:/Users/etorreborre/projects/specs2/specs2-fp-lib-on-js/fp/shared/src/main/scala/org/specs2/fp/Name.scala:30:30)
[error]   org.specs2.fp.Need$$anon$2.value(/Users/kenji/specs2-4/target/scala-2.12/file:/Users/etorreborre/projects/specs2/specs2-fp-lib-on-js/fp/shared/src/main/scala/org/specs2/fp/Name.scala:31:19)
[error]   {anonymous}()(/Users/kenji/specs2-4/target/scala-2.12/file:/Users/etorreborre/projects/specs2/specs2-fp-lib-on-js/common/shared/src/main/scala/org/specs2/control/eff/SafeEffect.scala:130:40)
```

examples
- https://github.com/scalaz/scalaz/blob/02f50d88fa9f8e8f83b99d33a3216b56137872f8/project/build.scala#L59-L63
- https://github.com/typelevel/cats/blob/68a8666ce7c84604835d8457c7c7c507d8cf2283/build.sbt#L73-L80
- https://github.com/milessabin/shapeless/blob/b7f87fd907f6d0dd182b2cf50d92f5e0422a5de8/build.sbt#L77-L84

**Comment by etorreborre** [comment]
Thanks for both the issues you opened, I will take care of that this week.

**Comment by etorreborre** [comment]
Both issues available as `4.0.0-RC2` in a few minutes.
